src/icon/Icon.ts

```diff
@@ -0,0 +1,59 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, ThemedProperties, theme } from '@dojo/widget-core/mixins/Themed';
+import { v } from '@dojo/widget-core/d';
+import { CustomAriaProperties } from '../common/interfaces';
+import { formatAriaProperties } from '../common/util';
+import * as iconCss from '../theme/common/icons.m.css';
+import { customElement } from '@dojo/widget-core/decorators/customElement';
+
+export type IconType = keyof typeof iconCss;
+
+/**
+ * @type IconProperties
+ *
+ * Properties that can be set on an Icon component
+ *
+ * @property type           Icon type, e.g. downIcon, searchIcon, etc.
+ */
+export interface IconProperties extends ThemedProperties, CustomAriaProperties {
+	type: IconType;
+	onClick?(): void;
+}
+
+export const ThemedBase = ThemedMixin(WidgetBase);
+
+@theme(iconCss)
+@customElement<IconProperties>({
+	tag: 'dojo-icon',
+	properties: [
+		'theme',
+		'aria',
+		'extraClasses'
+	],
+	attributes: [ 'type' ],
+	events: [
+		'onClick'
+	]
+})
+export class IconBase<P extends IconProperties = IconProperties> extends ThemedBase<P, null> {
+	private _onClick (event: MouseEvent) {
+		event && event.stopPropagation();
+		this.properties.onClick && this.properties.onClick();
+	}
+
+	protected render(): DNode {
+		const {
+			aria = {},
+			type
+		} = this.properties;
+
+		return v('i', {
+			...formatAriaProperties(aria),
```

Comment:
It's good to include the aria props, but it might be worthwhile to also have an explicit property for hidden descriptive text, maybe like `altText`? If it's provided, we can leave off `role="presentation"` and include a visually hidden span with the text. If it's not provided, we can include `role="presentation"` and `aria-hidden="true"`

Reply:
Now it's always setting `aria-hidden="true"` on the icon itself (unless explicitly overridden), never setting `role="presentation"` on anything, and includes a visually hidden with `altText` if provided.